src/Compilers/CSharp/Test/Syntax/Parsing/ForStatementParsingTest.cs

```diff
@@ -448,4 +443,2690 @@ public void TestVariableDeclaratorVersusCondition1()
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "b");
+                    }
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.SemicolonToken);
+            N(SyntaxKind.TupleExpression);
+            {
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.Argument);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "a");
+                    }
+                }
+                N(SyntaxKind.CommaToken);
+                N(SyntaxKind.Argument);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "b");
+                    }
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CloseParenToken);
+            N(SyntaxKind.EmptyStatement);
+            {
+                N(SyntaxKind.SemicolonToken);
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void TestVariousExpressions_Typeof()
+    {
+        UsingStatement("""
+            for (typeof(int);typeof(int);typeof(int));
+            """);
+
+        N(SyntaxKind.ForStatement);
+        {
+            N(SyntaxKind.ForKeyword);
+            N(SyntaxKind.OpenParenToken);
+            N(SyntaxKind.TypeOfExpression);
+            {
+                N(SyntaxKind.TypeOfKeyword);
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.PredefinedType);
+                {
+                    N(SyntaxKind.IntKeyword);
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.SemicolonToken);
+            N(SyntaxKind.TypeOfExpression);
+            {
+                N(SyntaxKind.TypeOfKeyword);
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.PredefinedType);
+                {
+                    N(SyntaxKind.IntKeyword);
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.SemicolonToken);
+            N(SyntaxKind.TypeOfExpression);
+            {
+                N(SyntaxKind.TypeOfKeyword);
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.PredefinedType);
+                {
+                    N(SyntaxKind.IntKeyword);
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CloseParenToken);
+            N(SyntaxKind.EmptyStatement);
+            {
+                N(SyntaxKind.SemicolonToken);
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void TestVariousExpressions_With1()
+    {
+        UsingStatement("""
+            for (a with { }; a with { }; a with { })
```

Comment:
similar issue, parser sees this as a var decl for `a with` with `a` as the type and `with` as the name.  As this is vastly unlikely to be ever written, fixing this seems unnecessary.